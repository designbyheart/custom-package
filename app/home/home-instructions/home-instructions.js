// @flow

import React, { PureComponent } from 'react'
import {
  ImageBackground,
  Image,
  StyleSheet,
  Linking,
  Text,
  View,
} from 'react-native'
import Snackbar from 'react-native-snackbar'

import { verticalScale, moderateScale } from 'react-native-size-matters'

import type { HomeInstructionsProps } from './type-home-instructions'
import { colors, fontFamily } from '../../common/styles/constant'

export class HomeInstructions extends PureComponent<
  HomeInstructionsProps,
  void
> {
  render() {
    const {
      headline,
      title,
      prodNetworkText,
      devNetworkText,
      usingProductionNetwork,
    } = this.props
    return (
      <ImageBackground
        source={require('../../images/connection-items-placeholder.png')}
        style={styles.backgroundImage}
      >
        <View style={[styles.contentContainer]}>
          <View>
            <Image
              source={require('../../images/CredentialGraphic.png')}
              resizeMode="contain"
              style={styles.image}
            />
            <Text style={[styles.headline]}>{headline}</Text>
            <Text style={[styles.title]}>{title}</Text>
            {usingProductionNetwork && (
              <Text style={[styles.text]}>{prodNetworkText}</Text>
            )}
            {!usingProductionNetwork && (
              <Text style={[styles.text]}>{devNetworkText}</Text>
            )}
          </View>
        </View>
      </ImageBackground>
    )
  }

  openTryConnectMe = () => {
    Linking.openURL('https://try.connect.me').catch(() => {
      Snackbar.show({
        text:
          'Could not open Tutorial website. No capable browser found to open it.',
        duration: Snackbar.LENGTH_LONG,
        backgroundColor: colors.cmRed,
        textColor: colors.cmWhite,
      })
    })
  }
}

const styles = StyleSheet.create({
  backgroundImage: {
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  contentContainer: {
    marginTop: verticalScale(-56),
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  image: {
    height: moderateScale(90, 2.5),
    alignSelf: 'center',
  },
  headline: {
    fontSize: moderateScale(20, 0.1),
    color: colors.cmOrange,
    marginVertical: moderateScale(5),
    fontFamily: fontFamily,
    textAlign: 'center',
    fontWeight: 'bold',
  },
  title: {
    fontSize: moderateScale(17, 0.1),
    marginVertical: moderateScale(5),
    color: colors.cmGray1,
    textAlign: 'center',
    fontFamily: fontFamily,
  },
  text: {
    fontSize: moderateScale(15, 0.1),
    marginVertical: moderateScale(5),
    marginHorizontal: '5%',
    color: colors.cmGray1,
    textAlign: 'center',
    fontFamily: fontFamily,
  },
})
