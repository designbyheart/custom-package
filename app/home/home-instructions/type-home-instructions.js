// @flow

export type HomeInstructionsProps = {
    headline: string,
    title: string,
    prodNetworkText: string,
    devNetworkText: string,
    usingProductionNetwork: boolean,
  }