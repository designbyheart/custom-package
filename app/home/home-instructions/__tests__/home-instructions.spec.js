// @flow

import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { HomeInstructions } from '../home-instructions'

describe('<HomeInstructions />', () => {
  function getProps(usingProductionNetwork: boolean) {
    return {
      headline: 'Headline',
      title: 'Title',
      prodNetworkText: 'Production network message',
      devNetworkText: 'Dev network message',
      usingProductionNetwork: usingProductionNetwork,
    }
  }

  it('should match snapshot', () => {
    const component = renderer.create(<HomeInstructions {...getProps(true)} />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should match snapshot for test network', () => {
    const component = renderer.create(<HomeInstructions {...getProps(false)} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
