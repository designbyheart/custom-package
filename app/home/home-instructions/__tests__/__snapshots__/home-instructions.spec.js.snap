// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeInstructions /> should match snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "default": "RelativeImageStub",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "marginTop": -44.470588235294116,
          "width": "100%",
        },
      ]
    }
  >
    <View>
      <Image
        resizeMode="contain"
        source={
          Object {
            "default": "RelativeImageStub",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 70.7142857142857,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#EB9B2D",
              "fontFamily": "Lato",
              "fontSize": 19.82857142857143,
              "fontWeight": "bold",
              "marginVertical": 4.785714285714286,
              "textAlign": "center",
            },
          ]
        }
      >
        Headline
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#505050",
              "fontSize": 16.854285714285716,
              "marginVertical": 4.785714285714286,
              "textAlign": "center",
            },
          ]
        }
      >
        Title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#505050",
              "fontSize": 14.871428571428572,
              "marginHorizontal": "5%",
              "marginVertical": 4.785714285714286,
              "textAlign": "center",
            },
          ]
        }
      >
        Production network message
      </Text>
    </View>
  </View>
</View>
`;

exports[`<HomeInstructions /> should match snapshot for test network 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "default": "RelativeImageStub",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "marginTop": -44.470588235294116,
          "width": "100%",
        },
      ]
    }
  >
    <View>
      <Image
        resizeMode="contain"
        source={
          Object {
            "default": "RelativeImageStub",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 70.7142857142857,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#EB9B2D",
              "fontFamily": "Lato",
              "fontSize": 19.82857142857143,
              "fontWeight": "bold",
              "marginVertical": 4.785714285714286,
              "textAlign": "center",
            },
          ]
        }
      >
        Headline
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#505050",
              "fontSize": 16.854285714285716,
              "marginVertical": 4.785714285714286,
              "textAlign": "center",
            },
          ]
        }
      >
        Title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#505050",
              "fontSize": 14.871428571428572,
              "marginHorizontal": "5%",
              "marginVertical": 4.785714285714286,
              "textAlign": "center",
            },
          ]
        }
      >
        Dev network message
      </Text>
    </View>
  </View>
</View>
`;
